### ISSUE TITLE
Value of 0 shown in Submit Transaction modal when value of 1 entered for extrinsic when first loaded

### ISSUE THREAD

**Issue by ltfschoen** [polkadot-js/apps#505]
In the "next" branch, the first time you go to http://localhost:3000/#/extrinsics to submit an extrinsic and perform a balances transfer without changing the default value, which is 1 (as defined in [packages/ui-app/src/Params/initValue.ts](https://github.com/polkadot-js/apps/blob/next/packages/ui-app/src/Params/initValue.ts#L25) for `Balance`), as shown below, when you click "Submit Transaction", the modal opens, **the value shown for the Balance in the modal is 0 instead of the default value of 1** (see second screenshot).

If you change the value of the 'value: Compact<Balance>' to different value other than the default value, then it will show the value correctly in the modal when you click "Submit Transaction".

When the http://localhost:3000/#/extrinsics is initially loaded, `onChange` of the Params component is called with that contains this param that has a `value` of 1
https://github.com/polkadot-js/apps/blob/next/packages/app-extrinsics/src/Extrinsic.tsx#L112, the value is stored in an `extrinsic` and passed to the `nextState` function and then as an argument of to the `onChange` prop, which is then stored in `extrinsic` of the Selection component's state https://github.com/polkadot-js/apps/blob/next/packages/app-extrinsics/src/Selection.tsx#L55. If we debug and show the value of the `extrinsic` argument passed to the `onChangeExtrinsic` function in the `Selection` component when the page loads, we can see that the value of `1` is stored as the value of the `words` property (i.e. `words: [1]`) of the "method" mapping
<img width="498" alt="screenshot 2018-12-11 at 09 34 57" src="https://user-images.githubusercontent.com/6226175/49787880-1e14d100-fd28-11e8-82d8-f868f6d06b7a.png">. If we click "Submit Transaction", it calls the `onQueue` function, which retrieves the value of `extrinsic` that's been stored in the `Selection` component's state, but  if we debug the value of `extrinsic` with a console.log here, it's value contains `0` instead of `1`  (i.e. `words: [0]`) as shown below https://github.com/polkadot-js/apps/blob/next/packages/app-extrinsics/src/Selection.tsx#L116
<img width="1158" alt="screenshot 2018-12-11 at 09 42 58" src="https://user-images.githubusercontent.com/6226175/49788256-27eb0400-fd29-11e8-9f97-62b72458de7b.png">

<img width="1175" alt="screenshot 2018-12-11 at 09 03 24" src="https://user-images.githubusercontent.com/6226175/49786310-9f1d9980-fd23-11e8-8a37-c132ec647839.png">

<img width="1192" alt="screenshot 2018-12-11 at 09 08 38" src="https://user-images.githubusercontent.com/6226175/49786506-5adec900-fd24-11e8-8312-66c1d72d9b19.png">

**Comment by ltfschoen** [comment]
this issue is happening with other extrinsic selections too. it appears to be caused because when the page loads, [`componentDidUpdate`](https://github.com/polkadot-js/apps/blob/next/packages/ui-app/src/Params/index.tsx#L81) is called multiple times, and if the previous state doesn't match the new state then it calls `onChange` to update the value of the extrinsic stored in the state, and the values in the modal use this extrinsic value.
We can observe and compare the value of `prevState.values` and `state.values` in packages/ui-app/src/Params/index.tsx#L81, i.e. `console.log(this.state.values[1].value.words[0])` and `console.log(prevState.values[1].value.words[0])`
Initially when it's loading the value of new state and previous state are:
  **prev. state:** undefined
  **new state:** value: { words: [1]
Then it changes to:
  **prev. state:** value: { words: [1]
  **new state:** value: { words: [1]
And then finally it changes to:
Then it changes to:
  **prev. state:** value: { words: [1]
  **new state:** value: { words: [0, empty]

Where the `0` in `words[0]` corresponds to the value of the label label "value: Compact<Balance>") that'll get shown in the modal when you click "Submit Transaction", even though before clicking the button the UI shows a value of `1`.

**Comment by jacogr** [comment]
Closed in #506, aligning InputNumber & Balance across the board. All Balances are now always starting at 0 - no fighting between defaults and would-be defaults. (Additionally doing transfer in the Extrinsics app now behaves the same as in the Transfer app as defaults go)

DefaultValue and Value and changed value with formatting is not easy to do elegantly (and InputNumber/Balance is anything but elegant with all the logic in there as it stands) since we don't want to not switch between controlled and uncontrolled inputs. We actually had this despite the React warnings that used to clog up the console.

TL:DR - Took the easy way of just aligning everything to a 0 default, the_perceived_ benefit 1 gives is basically non-existent, in both cases the number is way below existential amount and will be edited anyway.
